src/context/typecore.ml

```diff
@@ -481,6 +481,16 @@ let is_removable_field com f =
 			| _ -> false)
 	)
 
+let is_forced_inline c cf =
+	match c with
+	| Some { cl_kind = KAbstractImpl _ } -> true
+	| Some c when has_class_flag c CExtern -> true
+	| _ when has_class_field_flag cf CfExtern -> true
+	| _ -> false
+
+let needs_inline ctx c cf =
+	cf.cf_kind = Method MethInline && ctx.allow_inline && (is_forced_inline c cf || ctx.g.doinline)
```

Comment:
Shouldn't the `is_forced_inline` also trump `ctx.allow_inline = false`?

Reply:
Thought so too but it didn't before :/

Reply:
Hmm ok, then let's not change it for now.

But one minor thing: it seems sensible to check `doinline` before `is_forced_inline` to avoid the matching.